samples/Samples.Jwt/JwtWebSocketAuthenticationService.cs

```diff
@@ -0,0 +1,66 @@
+using Microsoft.AspNetCore.Authentication.JwtBearer;
+
+namespace GraphQL.Server.Samples.Jwt;
+
+/// <summary>
+/// Authenticates WebSocket connections via the 'payload' of the initialization packet.
+/// This is necessary because WebSocket connections initiated from the browser cannot
+/// authenticate via HTTP headers.
+/// <br/><br/>
+/// This class is not used when authenticating over GET/POST.
+/// </summary>
+public class JwtWebSocketAuthenticationService : IWebSocketAuthenticationService
+{
+    private readonly IGraphQLSerializer _graphQLSerializer;
+
+    public JwtWebSocketAuthenticationService(IGraphQLSerializer graphQLSerializer)
+    {
+        _graphQLSerializer = graphQLSerializer;
+    }
+
+    public Task AuthenticateAsync(IWebSocketConnection connection, string subProtocol, OperationMessage operationMessage)
+    {
+        try
+        {
+            // for connections authenticated via HTTP headers, no need to reauthenticate
+            if (connection.HttpContext.User.Identity?.IsAuthenticated ?? false)
+                return Task.CompletedTask;
+
+            // attempt to read the 'Authorization' key from the payload object and verify it contains "Bearer: XXXXXXXX"
+            var authPayload = _graphQLSerializer.ReadNode<AuthPayload>(operationMessage.Payload);
+            if (authPayload != null && authPayload.Authorization != null && authPayload.Authorization.StartsWith("Bearer ", StringComparison.Ordinal))
+            {
+                // pull the token from the value
+                var token = authPayload.Authorization.Substring(7);
+                // parse the token in the same manner that the .NET AddJwtBearer() method does
+                var handler = new Microsoft.IdentityModel.JsonWebTokens.JsonWebTokenHandler();
+                var result = handler.ValidateToken(token, JwtHelper.Instance.TokenValidationParameters);
+                if (result.IsValid)
+                {
+                    // convert JWT tokens with "role" as the claim type (a JWT defined claim type) to the proper .NET claim type constant
+                    // note this conversion automatically happens within the .NET AddJwtBearer() method when authenticating GET/POST requests
+                    var claims = result.ClaimsIdentity.Claims.Select(claim => claim.Type == "role" ? new Claim(result.ClaimsIdentity.RoleClaimType, claim.Value) : claim);
+                    var principal = new ClaimsPrincipal(new ClaimsIdentity(claims, JwtBearerDefaults.AuthenticationScheme));
+
+                    // set the ClaimsPrincipal for the HttpContext; authentication will take place against this object
+                    connection.HttpContext.User = principal;
+                }
+            }
+        }
+        catch
+        {
+            // no errors during authentication should throw an exception
+            // specifically, attempting to validate an invalid JWT token will result in an exception, which should be ignored
```

Comment:
I think this is a good case to use logging in example - inject MS logger and use it here to track invalid JWT tokens.

Reply:
Well, I wouldn't for production use any more than I'd use `ILogger` to track every invalid username / password attempted on the site.

Again, this is just sample code.  If you think adding an logger to it would benefit the sample, we could do so.

Reply:
I did try to make the sample somewhat useful for production applications, as it seems that many 'sample implementations' get used in production, as can be seen by the number of times we are asked questions regarding the GraphQL.NET Harness sample's middleware for ASP.NET Core.

Reply:
I posted a suggestion regarding comment.